compiler-api/src/main/java/com/squareup/anvil/compiler/api/AnvilApplicabilityChecker.kt

```diff
@@ -0,0 +1,9 @@
+package com.squareup.anvil.compiler.api
+
+public interface AnvilApplicabilityChecker {
```

Comment:
The goal with this is to be able to hoist up the applicability checks to shared locations